notebook/static/notebook/js/keyboardmanager.js

```diff
@@ -25,24 +25,32 @@ define([
          * @class KeyboardManager
          * @constructor
          * @param options {dict} Dictionary of keyword arguments :
-         *    @param options.events {$(Events)} instance 
+         *    @param options.events {$(Events)} instance
          *    @param options.pager: {Pager}  pager instance
          */
         this.mode = 'command';
+        this.view = 'tree';
         this.enabled = true;
         this.pager = options.pager;
         this.quick_help = undefined;
         this.notebook = undefined;
+        this.notebooklist = undefined;
         this.last_mode = undefined;
         this.bind_events();
         this.env = {pager:this.pager};
         this.actions = options.actions;
-        this.command_shortcuts = new keyboard.ShortcutManager(undefined, options.events, this.actions, this.env );
-        this.command_shortcuts.add_shortcuts(this.get_default_common_shortcuts());
-        this.command_shortcuts.add_shortcuts(this.get_default_command_shortcuts());
-        this.edit_shortcuts = new keyboard.ShortcutManager(undefined, options.events, this.actions, this.env);
-        this.edit_shortcuts.add_shortcuts(this.get_default_common_shortcuts());
-        this.edit_shortcuts.add_shortcuts(this.get_default_edit_shortcuts());
+        if (this.view === 'tree'){
+          this.tree_shortcuts = new keyboard.ShortcutManager(undefined, options.events, this.actions, this.env);
+          this.tree_shortcuts.add_shortcuts(this.get_default_tree_shortcuts());
+        }
+        if (this.view === 'notebook'){
```

Comment:
the way things update, this.view updates the way you want it to. you won't
run into problems

On Tue, Aug 4, 2015 at 5:48 PM, Feynman Liang <notifications@github.com>
wrote:

> In notebook/static/notebook/js/keyboardmanager.js
> <https://github.com/jupyter/notebook/pull/217#discussion_r36259861>:
>
> >          this.last_mode = undefined;
> >          this.bind_events();
> >          this.env = {pager:this.pager};
> >          this.actions = options.actions;
> > -        this.command_shortcuts = new keyboard.ShortcutManager(undefined, options.events, this.actions, this.env );
> > -        this.command_shortcuts.add_shortcuts(this.get_default_common_shortcuts());
> > -        this.command_shortcuts.add_shortcuts(this.get_default_command_shortcuts());
> > -        this.edit_shortcuts = new keyboard.ShortcutManager(undefined, options.events, this.actions, this.env);
> > -        this.edit_shortcuts.add_shortcuts(this.get_default_common_shortcuts());
> > -        this.edit_shortcuts.add_shortcuts(this.get_default_edit_shortcuts());
> > +        if (this.view === 'tree'){
> > +          this.tree_shortcuts = new keyboard.ShortcutManager(undefined, options.events, this.actions, this.env);
> > +          this.tree_shortcuts.add_shortcuts(this.get_default_tree_shortcuts());
> > +        }
> > +        if (this.view === 'notebook'){
>
> else if
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/217/files#r36259861>.
>